sdk/storage/azure-storage-file-datalake/out/azure/storage/filedatalake/_shared_access_signature.pyi

```diff
@@ -0,0 +1,10 @@
+# Stubs for azure.storage.filedatalake._shared_access_signature (Python 3)
+#
+# NOTE: This dynamically typed stub was automatically generated by stubgen.
+
+from typing import Any, Optional
+
+def generate_account_sas(account_name: str, account_key: str, resource_types: Union[ResourceTypes, str], permission: Union[AccountSasPermissions, str], expiry: Optional[Union[datetime, str]], start: Optional[Union[datetime, str]]=..., ip: Optional[str]=..., **kwargs: Any) -> str: ...
+def generate_file_system_sas(account_name: str, file_system_name: str, account_key: Optional[str]=..., user_delegation_key: Optional[UserDelegationKey]=..., permission: Optional[Union[FileSystemSasPermissions, str]]=..., expiry: Optional[Union[datetime, str]]=..., start: Optional[Union[datetime, str]]=..., ip: Optional[str]=..., **kwargs: Any) -> str: ...
+def generate_directory_sas(account_name: str, file_system_name: str, directory_name: str, account_key: Optional[str]=..., user_delegation_key: Optional[UserDelegationKey]=..., permission: Optional[Union[BlobSasPermissions, str]]=..., expiry: Optional[Union[datetime, str]]=..., start: Optional[Union[datetime, str]]=..., ip: Optional[str]=..., **kwargs: Any) -> str: ...
+def generate_file_sas(account_name: str, file_system_name: str, directory_name: str, file_name: str, account_key: Optional[str]=..., user_delegation_key: Optional[UserDelegationKey]=..., permission: Optional[Union[BlobSasPermissions, str]]=..., expiry: Optional[Union[datetime, str]]=..., start: Optional[Union[datetime, str]]=..., ip: Optional[str]=..., **kwargs: Any) -> str: ...
```

Comment:
I'm sure we can move more of the positional params in these functions into kwargs as well.

Reply:
Possible suggestion:
```
def generate_file_sas(
    account_name: str,
    file_system_name: str,
    directory_name: str, 
    file_name: str, 
    credential: str,
    credential_type=ACCOUNT_KEY, # or USER_DELEGATION_KEY
    permission: Optional[Union[BlobSasPermissions, str]]=..., 
    expiry: Optional[Union[datetime, str]]=..., 
       **kwargs: Any)
```

Reply:
I like this suggestion, but this would make it inconsistent with what we have in blobs, queues and fileshares. That is something we might want to consider.